src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,561 @@
+
+        if not np.allclose(channel_locations_rel, templates_locations_rel):
+            warnings.warn("Channel locations are different between recording and templates. Interpolating templates.")
+            templates_array = np.zeros(templates.templates_array.shape, dtype=dtype)
+            for i in range(len(templates_array)):
+                src_template = templates.templates_array[i][np.newaxis, :, :]
+                templates_array[i] = interpolate_templates(src_template, templates_locations_rel, channel_locations_rel)
+        else:
+            templates_array = templates.templates_array
+
+        # manage scaling of templates
+        templates_ = templates
+        if recording.has_scaleable_traces():
+            if templates_in_uV:
+                templates_array = (templates_array - recording.get_channel_offsets()) / recording.get_channel_gains()
+                # make a copy of the templates and reset templates_array (might have scaled templates)
+                templates_ = templates.select_units(templates.unit_ids)
+                templates_.templates_array = templates_array
+
+    if sorting is None:
+        generate_sorting_kwargs = generate_sorting_kwargs.copy()
+        generate_sorting_kwargs["durations"] = durations
+        generate_sorting_kwargs["sampling_frequency"] = sampling_frequency
+        generate_sorting_kwargs["seed"] = seed
+        sorting = generate_sorting(**generate_sorting_kwargs)
+    else:
+        assert sorting.sampling_frequency == sampling_frequency
+
+    num_spikes = sorting.to_spike_vector().size
+    sorting.set_property("gt_unit_locations", unit_locations)
+
+    assert (nbefore + nafter) == templates_array.shape[
+        1
+    ], "templates and ms_before, ms_after should have the same length"
+
+    if templates_array.ndim == 3:
+        upsample_vector = None
+    else:
+        if upsample_vector is None:
+            upsample_factor = templates_array.shape[3]
+            upsample_vector = rng.integers(0, upsample_factor, size=num_spikes)
+
+    if amplitude_std is not None:
+        amplitude_factor = rng.normal(loc=1, scale=amplitude_std, size=num_spikes)
+    else:
+        amplitude_factor = None
+
+    if motion is not None:
+        assert num_segments == motion.num_segments, "recording and motion should have the same number of segments"
+        dim = motion.dim
+        motion_array_concat = np.concatenate(motion.displacement)
+        if dim == 0:
+            start = np.array([np.min(motion_array_concat), 0])
+            stop = np.array([np.max(motion_array_concat), 0])
+        elif dim == 1:
+            start = np.array([0, np.min(motion_array_concat)])
+            stop = np.array([0, np.max(motion_array_concat)])
+        elif dim == 2:
+            raise NotImplementedError("3D motion not implemented yet")
+        displacements = make_linear_displacement(start, stop, num_step=int((stop - start)[dim]))
```

Comment:
Should we expose an option for the num_step ?
here we are forcing a 1um step, this could be hudge, am i wrong ?